src/main/java/racingcar/validation/CarsValidator.java

```diff
@@ -0,0 +1,35 @@
+package racingcar.validation;
+
+import static racingcar.validation.Exceptions.DUPLICATED_CAR_NAME;
+import static racingcar.validation.Exceptions.INSUFFICIENT_CARS;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import racingcar.domain.Car;
+
+public class CarsValidator implements Validator<List<Car>> {
+    private static final int MIN_CARS_AMOUNT = 2;
+
+    @Override
+    public void validate(List<Car> cars) {
+        validateNameDuplication(cars);
+        validateCarsSize(cars);
+    }
+
+    private void validateNameDuplication(List<Car> cars) {
+        Set<String> uniqueCarNames = new HashSet<>();
+        for (Car car : cars) {
+            String lowerCarName = car.getName().toLowerCase();
+            if (!uniqueCarNames.add(lowerCarName)) {
```

Comment:
if문의 조건은 가독성의 측면에서 `!`을  쓰지 않는 `긍정문`을 권장하더라구요! 해당 부분을 메서드로 분리하면 긍정문을 유도할 수 있지 않을까요?

```suggestion
            if(canNotAdd(uniqueCarNames, lowerCarName){
                ....
            }
             
            private boolean canNotAdd(Set<> names, String lowerName) {
                return !names.add(lowerName);
            }
```

Reply:
아하..!! 저도 if문에 부정연산자를 최대한 지양해야 하는 건 알고 있었는데요, 메서드로 따로 빼서 구현해야 할 만큼 가독성을 해치지는 않는다고 생각해 그냥 두었습니다..!
오히려 메서드를 분리하면 시선이 분산되어 가독성이 떨어지지 않을까 생각도 들었구요.
이부분 kkonii님은 어떻게 생각하시나요??

Reply:
맞아요 작성하신 코드가 읽기에는 불편함이 없습니다! 
저는 클린 코드 학습할 당시에 `메서드명` + `!` 이면 읽는 사람이 두 번 생각하게 된다는 말에 정말 많이 공감했어서, 항상 `이걸 모르는 사람은 없겠지만... 이왕이면 조금 더 확실하게 ` 라는 생각으로 바라보는 거 같아요ㅋㅋㅋ

Reply:
오 읽는 사람이 두 번 생각하게 한다라..!! 맞는 말이네요 ㅋㅋㅋㅋ
메서드를 분리해서 시선이 분산될 것 같으면, 분리된 메서드까지 시선이 안가도 메서드명만으로 조건이 이해가 되도록 만들면 되니까요 ㅎㅎ 읽는 사람의 기준에서 생각해보는 연습 더 해야겠습니다.
가독성을 높이려고 많이 노력했는데 이런 사소한 부분이 아직 멀었네요. 좋은 말씀 감사합니다!